### DISCUSSION TITLE
LaTeX figure output proposal & discussion.

### DISCUSSION THREAD

**Discussion by argent0** [tarleb/pandoc#23] [Q&A]
I'm repeating this in a different thread to encourage discussion.

There exist the packages `caption` and `subcaption` that enable [subfigures](https://www.overleaf.com/learn/latex/How_to_Write_a_Thesis_in_LaTeX_(Part_3):_Figures,_Subfigures_and_Tables#Subfigures)

<details>
<summary> Click to expand example </summary>

```{latex}
\usepackage{caption}
\usepackage{subcaption}
...
\begin{figure}
     \begin{subfigure}[b]{0.5\textwidth}
		Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod

     	\begin{subfigure}[b]{0.5\textwidth}
			\centering
			This.
         	\caption{Inner 1.1}
     	\end{subfigure}
     	\begin{subfigure}[b]{0.5\textwidth}
			\centering
			That.
         	\caption{Inner 1.2}
     	\end{subfigure}
         \caption{Inner 1}
         \label{fig:inner1}
     \end{subfigure}
     \begin{subfigure}[b]{0.5\textwidth}
		Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
         \caption{$Inner 2$}
         \label{fig:inner2}
     \end{subfigure}
        \caption{Outer caption}
        \label{fig:outer}
\end{figure}
```

</details>

The `caption` package plays well with latex's default [classes](https://tex.stackexchange.com/questions/34579/is-there-really-something-wrong-with-using-the-caption-package-for-continuedflo)(book, report, article) and `beamer` and `thesis`.

But is not installed in the default ubuntu package. It is on ubuntu on this [package](https://packages.ubuntu.com/bionic/all/texlive-latex-extra/filelist).

One idea would be to include figure support as an extension, in this way you would only get sub figure support when you require it (but would still need the `texlive-extra` package installed.

## Custom floats

The previous `figure` and `subfigure` seem to specifically represent images.

It's also possible to define custom [floats](https://en.wikibooks.org/wiki/LaTeX/Floats,_Figures_and_Captions#Custom_floats), but again it is important to note that those can't be nested.

## About nesting

Ideally we would like to have something like:

```{latex}
\begin{foo}
	...arbitrary latex content ...
\end{foo}
```

Unfortunately LaTeX doesn't allow this in most cases. So I see two alternatives:

1. We limit nesting in the output: We set a limit to some N. But later latex
   versions may change in behavior requiring a pandoc update.

2. We produce arbitrarily nested output. That, although semantically correct,
   isn't valid LaTeX.

**Comment by cderv** [comment]
I think there is also other option like Subfig package (https://www.ctan.org/pkg/subfig). We use that in rmarkdown when doing multiple figure. Example in: https://bookdown.org/yihui/rmarkdown-cookbook/latex-subfigure.html
However caption would still be needed I think and the limitation of the available package still remains. 

About nested output, at some point, **minipage** package needs to be used to organize sub figures correctly. 
Something like this for several figures
````latex
\begin{figure}
\begin{minipage}[t]{0.50\linewidth}
{\centering 
\raisebox{-\height}{
\includegraphics{plot1.png}
}
\caption{plot 1}
}
\end{minipage}%
%
\begin{minipage}[t]{0.50\linewidth}
{\centering 
\raisebox{-\height}{
\includegraphics{plot1.png}
}
\caption{plot 1 again}
}
\end{minipage}%
\newline
\begin{minipage}[t]{1.00\linewidth}
{\centering 
\raisebox{-\height}{
\includegraphics{plot1.png}
}
\caption{Plot 1 again as I have only this one}
}
\end{minipage}%
\end{figure}
````

or like this using subfloat with main caption

````latex
\begin{figure}
\subfloat[A plot]{\label{fig11}%
\begin{minipage}[t]{0.50\linewidth}
\raisebox{-\height}{
{\centering 
\includegraphics{plot1.png}
}
}\end{minipage}%
}
%
\subfloat[Another plot]{\label{fig12}%
\begin{minipage}[t]{0.50\linewidth}
\raisebox{-\height}{
{\centering 
\includegraphics{plot1.png}
}
}\end{minipage}%
}
\caption{\label{fig1}Some plots}
\end{figure}
````

Does that help ? 

I am not sure if I understood correctly where we are at currently . 😄

**Reply by argent0** [reply]
Yes this is very helpful. I think the first iteration would limit the depth to 2 and flatten anything deeper.
